docs/language/README.md

```diff
@@ -1,35 +1,268 @@
-# Zed Language
+# The Zed Language
 
-The Zed system includes a powerful language for searching, analyzing, and
-shaping data. The language embraces a _pipeline_ syntax that should be familiar
-to those who have worked with UNIX shells. Here's a simple example query:
+## TL;DR
+
+Zed is a pipeline-style search and analytics language for querying
+data in files, over HTTP, in S3 storage, or in a
+[Zed data lake](../lake).
+A simple Zed query has the following structure:
 
 ![Example Zed 1](images/example-zed.png)
 
 As is typical with pipelines, you can imagine the data flowing left-to-right
 through this chain of processing elements, such that the output of each element
-is the input to the next. The example above follows a common pattern seen in
+is the input to the next.  While Zed follows the common pattern seen in
 other query languages where the pipeline begins with a search and further
-processing is then performed on the isolated data. However, one of Zed's
+processing is then performed on the isolated data, one of Zed's
 strengths is that searches and expressions can appear in any order in the
 pipeline.
 
 ![Example Zed 2](images/example-zed-operator-search.png)
 
-The available pipeline elements are broadly categorized into:
+A complete list of [Zed operators](#operators) is found below.
+
+## Background
+
+An ambitious goal of the Zed project is to offer a language
+&mdash; the _Zed language_ &mdash;
+that provides an easy learning curve and a gentle slope from simple keyword search
+to log-search-style processing and ultimately to sophisticated, large-scale
+warehouse-scale queries.  The language also embraces a rich set of type operators
+based on the [Zed data model](../formats/zson.md) for data shaping
+to provide flexible and easy ETL.
+
+The simplest Zed program is perhaps a single word search, e.g.,
+```
+widget
+```
+This program searches the implied input for Zed records that
+contain the string "widget".
+
+> NOTE: we should clarify keyword search vs substring match.
+
+As with the unix shell and legacy log search systems,
+the Zed language embraces a _pipeline_ model where a source of data
+is treated as a stream then one or more operators concatenated with
+the `|` symbol transform, filter, and aggregate the stream, e.g.,
+```
+widget | price > 1000
+```
+
+That said, the Zed language is declarative and
```

Comment:
@nwt: I know you said you don't feel strongly, but I'll explain why I included it. One of the things @mccanne and I discussed offline about this this revision is that this intro text is more academic than our materials have been in the past. This seems fine, especially considering that some of the audiences we want to reach going forward are indeed more academic in nature. That said, we still have legacy users that may have learned what they know the language via the clicky features in Brim that effectively write it for them. If they follow the docs link in the app because they want to learn to do a new operation that's only _slightly_ more advanced than what they've already done, I get nervous if the first page they read gives the impression of "you must be _\<this>_ academic to ride the Zed ride". While almost all citizens at this point are conditioned to do web searches of unfamiliar terms, in keeping with the promise of a "gentle slope" learning curve, I felt like providing links for the more academic terms is a friendly gesture to these readers since it points them directly at resources should they want to keep climbing the slope. For me, a word like "declarative" is particularly tricky because it's generic enough that if a user just does a web search for it, the first hits they find may not be the ones that explain it in the context that relates to how it's used in Zed.

I'm absolutely open to swapping out the links to destinations deemed superior to Wikipedia.